spec/features/bulk_upload/bulk_add_trainees_spec.rb

```diff
@@ -154,14 +156,42 @@
         when_the_upload_has_failed_with_validation_and_duplicate_errors
         and_i_dont_see_that_the_upload_is_processing
         and_i_visit_the_summary_page(upload: @failed_upload)
-        then_i_see_the_review_page
+        then_i_see_the_review_page_with_no_errors
         and_i_dont_the_number_of_trainees_that_can_be_added
         and_i_see_the_validation_errors(number: 2)
         and_i_see_the_duplicate_errors(number: 3)
         and_i_see_the_review_errors_message
         and_i_see_the_review_errors_link
         and_i_dont_see_the_submit_button
       end
+
+      scenario "when I try to upload a file with errors" do
+        when_i_visit_the_bulk_update_index_page
+        and_i_click_the_bulk_add_trainees_page
+        then_i_see_how_instructions_on_how_to_bulk_add_trainees
+        and_i_see_the_empty_csv_link
+
+        when_i_attach_a_file_with_invalid_rows
+        and_i_click_the_upload_button
+        when_the_background_job_is_run
+        and_i_refresh_the_page
+        then_i_see_the_review_page_with_no_errors
+
+        when_the_background_job_is_run
+        and_i_refresh_the_page
+        then_i_see_the_review_page_with_validation_errors
+
+        when_i_click_the_review_errors_link
+        then_i_see_the_review_errors_page
+
+        when_i_click_on_the_download_link
+        then_i_receive_the_file
+
+        when_i_return_to_the_review_errors_page
+        and_i_attach_a_valid_file
+        and_i_click_the_upload_button
+        then_i_see_that_the_upload_is_processing
+      end
```

Comment:
I suggest to refactor the `the bulk add trainees page is visible` scenario instead to avoid running similar assertions twice.

Reply:
If it is unavoidable to have a separate scenario no worries.